src/Integrator/BDHI/PSE/NearField.cuh

```diff
@@ -28,6 +33,8 @@ namespace uammd{
 	  pg(pg),
 	  tolerance(par.tolerance)
 	{
```

Comment:
You are not setting lanczosTolerance here, meaning that the parameter that you added in BDHI is completely ignored here. Furthermore, you did not give this parameter a default value, which means any code using PSE currently would be broken.
Please add a default value for the parameter here, as:
```c++
  tolerance(par.tolerance),
  lanczosTolerance(par.lanczosTolerance){
  if(this->lanczosTolerance<0) this->lanczosTolerance = std::min(0.05f, sqrt(par.tolerance));
```
This way the old behavior is mantained if a user does not know about your new parameter.